lib/class-wp-rest-search-controller.php

```diff
@@ -0,0 +1,539 @@
+					'format'      => 'uri',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'title' => array(
+					'description' => __( 'The title for the object.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'properties'  => array(
+						'raw'      => array(
+							'description' => __( 'Title for the object, as it exists in the database.', 'gutenberg' ),
+							'type'        => 'string',
+							'context'     => array( 'edit' ),
+							'readonly'    => true,
+						),
+						'rendered' => array(
+							'description' => __( 'HTML title for the object, transformed for display.', 'gutenberg' ),
+							'type'        => 'string',
+							'context'     => array( 'view', 'edit', 'embed' ),
+							'readonly'    => true,
+						),
+					),
+				),
+				'type'  => array(
+					'description' => __( 'Type of the object.', 'gutenberg' ),
+					'type'        => 'string',
+					'enum'        => array_keys( $this->get_allowed_post_types() ),
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Retrieves the query params for the search results collection.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+
+		$query_params = parent::get_collection_params();
+
+		$query_params['context']['default'] = 'view';
+
+		$query_params['type'] = array(
+			'default'     => 'any',
+			'description' => __( 'Limit search results to content of one or more types.', 'gutenberg' ),
+			'type'        => 'array',
+			'items'       => array(
+				'enum' => array_merge( array_keys( $this->get_allowed_post_types() ), array( 'any' ) ),
+				'type' => 'string',
+			),
+		);
+
+		$query_params['type_exclude'] = array(
+			'default'     => array(),
+			'description' => __( 'Ensure search results exclude content of one or more specific types.', 'gutenberg' ),
+			'type'        => 'array',
+			'items'       => array(
+				'enum' => array_keys( $this->get_allowed_post_types() ),
+				'type' => 'string',
+			),
+		);
+
+		return $query_params;
+	}
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param WP_Post $post Post object.
+	 * @return array Links for the given post.
+	 */
+	protected function prepare_links( $post ) {
+
+		$links = array();
+
+		$item_route = $this->detect_rest_item_route( $post );
+		if ( ! empty( $item_route ) ) {
+			$links['self'] = array(
+				'href'       => rest_url( $item_route ),
+				'embeddable' => true,
+			);
+		}
+
+		$links['collection'] = array(
+			'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+		);
+
+		$links['about'] = array(
+			'href' => rest_url( 'wp/v2/types/' . $post->post_type ),
+		);
+
+		if ( ! empty( $post->post_author ) ) {
+			$links['author'] = array(
```

Comment:
Does `author` need to be included as a link if it's not in the response?

Reply:
No, makes sense to remove it.